dask/utils.py

```diff
@@ -625,3 +630,83 @@ def insert(tup, loc, val):
     L = list(tup)
     L[loc] = val
     return tuple(L)
+
+
+def infer_storage_options(urlpath, storage_options=None):
+    """ Infer storage options from URL path and merge it with existing storage
+    options.
+
+    Parameters
+    ----------
+    path: str or unicode
+        Either local absolute file path or URL (hdfs://namenode:8020/file.csv)
+    storage_options: dict (optional)
+        Its contents will get merged with the infered information from the
+        given path
+
+    Returns
+    -------
+    Storage options dict.
+
+    >>> so = infer_storage_options('/mnt/datasets/test.csv')
+    >>> so.pop('protocol')
+    'file'
+    >>> so.pop('path')
+    '/mnt/datasets/test.csv'
+    >>> so
+    {}
```

Comment:
I used them as valid doctests (and they pass). I will add an example section then.